### ISSUE TITLE
[BUG] Failed to update pull secret on the disk: Temporary error: pull secret not updated to disk 

### ISSUE THREAD

**Issue by oluies** [code-ready/crc#2627]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V 
  * Did you run `crc setup` before starting it (Yes/No)?yes
  * Running CRC on: Laptop

## CRC version
```bash
odeReady Containers version: 1.30.1+376cc3c1
OpenShift version: 4.8.2 (not embedded in executable)
```
  
## CRC status
```bash
crc status --log-level debug
DEBU HTTP-PROXY: http://klientproxy-nn.nn.se:8080, HTTPS-PROXY: http://klientproxy-nn.nn.se:8080, NO-PROXY: 127.0.0.1,localhost,.nn.se,.nnc.se,.nn.se, proxyCAFile: C:/WS/ocp-tools/konfiguration-klient-0.0.7/ca-certs/skv-ca-bundle
DEBU CodeReady Containers version: 1.30.1+376cc3c1
DEBU OpenShift version: 4.8.2 (not embedded in executable)
DEBU Running 'crc status'
DEBU Checking file: C:\Users\NN\.crc\machines\crc\.crc-exist
DEBU Checking file: C:\Users\NN\.crc\machines\crc\.crc-exist
DEBU Running '( Hyper-V\Get-VM crc ).state'
DEBU Running SSH command: df -B1 --output=size,used,target /sysroot | tail -1
DEBU Using ssh private keys: [C:\Users\NN\.crc\machines\crc\id_ecdsa C:\Users\NN\.crc\cache\crc_hyperv_4.8.2\id_ecdsa_crc]
DEBU SSH command results: err: <nil>, output: 32737570816 16710479872 /sysroot
DEBU cannot get OpenShift status: CreateFile C:\Users\NN\.crc\machines\crc\kubeconfig: Det går inte att hitta filen.
CRC VM:          Running
OpenShift:       Unreachable (v4.8.2)
Disk Usage:      16.71GB of 32.74GB (Inside the CRC VM)
Cache Usage:     59.49GB
Cache Directory: C:\Users\NN\.crc\cache
```

## CRC config
```bash
crc config view
- consent-telemetry                     : no
- cpus                                  : 8
- disk-size                             : 84
- http-proxy                            : http://klientproxy-nn.nn.se:8080
- https-proxy                           : http://klientproxy-nn.nn.se:8080
- memory                                : 30480
- nameserver                            : 10.167.0.100
- network-mode                          : vsock
- no-proxy                              : .nn.se,.nnc.se,.nn.se
- proxy-ca-file                         : C:/WS/ocp-tools/konfiguration-klient-0.0.7/ca-certs/nn-ca-bundle
- pull-secret-file                      : C:/WS/git/bin/pull-secret.txt
```

## Host Operating System
```bash

Host Name:                 NN
OS Name:                   Microsoft Windows 10 Enterprise
OS Version:                10.0.19042 N/A Build 19042
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Member Workstation
OS Build Type:             Multiprocessor Free
Product ID:                00329-00000-00003-AA985
Original Install Date:     2021-02-24, 14:20:16
System Boot Time:          2021-08-02, 17:27:38
System Manufacturer:       HP
System Model:              HP ZBook Firefly 15 G7 Mobile Workstation
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 12 GenuineIntel ~1803 Mhz
BIOS Version:              HP S73 Ver. 01.03.02, 2020-11-04
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume2
System Locale:             sv;Svenska
Input Locale:              sv;Svenska
Time Zone:                 (UTC+01:00) Amsterdam, Berlin, Bern, Rom, Stockholm, Wien
Total Physical Memory:     65 309 MB
Available Physical Memory: 59 401 MB
Virtual Memory: Max Size:  81 693 MB
Virtual Memory: Available: 75 798 MB
Virtual Memory: In Use:    5 895 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    NNN.SE
Hotfix(s):                 10 Hotfix(s) Installed.
                           [01]: KB5003537
                           [02]: KB4562830
                           [03]: KB4570334
                           [04]: KB4577586
                           [05]: KB4580325
                           [06]: KB4589212
                           [07]: KB4593175
                           [08]: KB4598481
                           [09]: KB5004237
                           [10]: KB5003742
```

### Steps to reproduce

  1. install... run setup tried several times
  2. get Failed to update pull secret on the disk: Temporary error: pull secret not updated to disk (x205)
  3. 
  4. 

### Logs

Before gather the logs try following if that fix your issue
```bash

DEBU Running SSH command: host -R 3 foo.apps-crc.testing
DEBU SSH command results: err: <nil>, output: foo.apps-crc.testing has address 192.168.127.2
INFO Check internal and public DNS query...
DEBU Running SSH command: host -R 3 quay.io
DEBU SSH command results: err: Process exited with status 1, output: Host quay.io not found: 3(NXDOMAIN)
WARN Failed public DNS query from the cluster: ssh command error:
command : host -R 3 quay.io
err     : Process exited with status 1\n : Host quay.io not found: 3(NXDOMAIN)
INFO Check DNS query from host...


EBU Running SSH command: <hidden>
DEBU SSH command succeeded
DEBU error: Temporary error: pull secret not updated to disk - sleeping 2s
DEBU retry loop: attempt 203
DEBU Running SSH command: <hidden>
DEBU SSH command succeeded
DEBU error: Temporary error: pull secret not updated to disk - sleeping 2s
DEBU retry loop: attempt 204
DEBU Running SSH command: <hidden>
DEBU SSH command succeeded
DEBU error: Temporary error: pull secret not updated to disk - sleeping 2s
DEBU RetryAfter timeout after 205 tries
Failed to update pull secret on the disk: Temporary error: pull secret not updated to disk (x205)
```

Please consider posting the output of `crc start --log-level debug`  on http://gist.github.com/ and post the link in the issue.

**Comment by gbraad** [comment]
You experience network issues in general.

Please note that `- nameserver                            : 10.167.0.100` has no effect when using the `vsock` network mode.

**Comment by gbraad** [comment]
it fails here:

https://github.com/code-ready/crc/blob/0a62918982b0fca88d8beae4f6e3602ce88b5788/pkg/crc/cluster/cluster.go#L452-L465

@praveenkumar any idea what could cause this?

**Comment by oluies** [comment]
> it fails here:
> 
> https://github.com/code-ready/crc/blob/0a62918982b0fca88d8beae4f6e3602ce88b5788/pkg/crc/cluster/cluster.go#L452-L465
> 
> @praveenkumar any idea what could cause this?

Latest fixpack just applied for Windows 10 2H2, could be that causing an issue. Will try with cable ethernet and disabled vpn etc.

**Comment by oluies** [comment]
*.apps-crc.testing works well, eg 
ping sole-openshift-console.apps-crc.testing works 
ping api.crc.testing works 

But: 
```
$ crc console --credentials
To login as a regular user, run 'oc login -u developer -p developer https://api.crc.testing:6443'.
To login as an admin, run 'oc login -u kubeadmin -p pass https://api.crc.testing:6443'
kubeadmin@api-crc-testing:6443 default .../git
$ oc login --insecure-skip-tls-verify=false -u kubeadmin -p pass  https://api.crc.testing:6443
Login failed (401 Unauthorized)
Verify you have provided correct credentials.
```

**Comment by oluies** [comment]
> You experience network issues in general:
> 
> ```
> WARN Failed public DNS query from the cluster: ssh command error:
> command : host -R 3 quay.io
> err     : Process exited with status 1\n : Host quay.io not found: 3(NXDOMAIN)
> INFO Check DNS query from host...
> ```
> 
> Please note that `- nameserver : 10.167.0.100` has no effect when using the `vsock` network mode.

I am behind a proxy, can I get it to use proxy for quay.io

**Comment by gbraad** [comment]
> 401

this means the API server is handling your request...

in that case it is something else. The cluster now starts, but might not be stable yet.

**Comment by gbraad** [comment]
> I am behind a proxy, can I get it to use proxy for quay.io, also we have a artifactory mirror

ignoring the question, as this is not the issue you filed for. please correct a separate discussion for that.

**Comment by oluies** [comment]
> > 401
> 
> this means the API server is handling your request...
> 
> in that case it is something else. The cluster now starts, but might not be stable yet.

developer works but not kubeadmin. I can login with the webconsole with developer but not kubeadmin.

**Comment by oluies** [comment]
I also get connection errors in the 

```net.Dial() = dial tcp 10.217.4.1:443: connectex: Det gick inte att göra en anslutning eftersom måldatorn aktivt nekade det.
net.Dial() = dial tcp 192.168.122.1:16509: connectex: Anslutningsförsöket misslyckades eftersom den anslutna datorn inte svarade inom en viss tid eller på grund av att den etablerade anslutningen till värddatorn inte längre fungerar.

PS C:\Users\YYJN> get-vm | ?{$_.State -eq "Running"} | select -ExpandProperty networkadapters | select vmname, macaddress, switchname, ipaddresses
PS C:\Users\YYJN> get-vm | ?{$_.State -eq "Running"}                                                                    
Name State   CPUUsage(%) MemoryAssigned(M) Uptime           Status           Version
---- -----   ----------- ----------------- ------           ------           -------
crc  Running 8           30480             00:07:11.7820000 Fungerar normalt 9.0


PS C:\Users\YYJN> get-vm | ?{$_.State -eq "Running"} | select -ExpandProperty networkadapters   
>
```

**Comment by gbraad** [comment]
messages like 192.168.122.1:16509 being recorded on the command line of the daemon are related to the cluster trying to connect to the libvirt environment. Where did you see this? It is not an issue for the operation of the cluster.

what does `crc status` give now?

Note that 401 might indicate on issue with the password you provided or the auth provider not reconciled yet.  Did you retry?

**Comment by praveenkumar** [comment]
@oluies When it failed for you first time `Failed to update pull secret on the disk: Temporary error: pull secret not updated to disk (x205)` did you deleted it again and tried it or you just did `crc start` and then trying with the `kubeadmin` password which is provided in the output?

What I am suspecting is you didn't delete and start but just did a start and because a new kubeadmin password is generated but it is not applied to required configmap you are getting this error with kubeadmin. `developer` user have fix password so it works as expected.

**Comment by oluies** [comment]
I did crc delete -f and cleanup

**Comment by gbraad** [comment]
Since developer worked I believe it would be best to perform the steps are

```
crc delete
crc start
```

Plesse, do not run `crc cleanup` as the side-effects are that you also need to re-setup the environment.

**Comment by oluies** [comment]
> Since developer worked I believe it would be best to perform the steps are
> 
> ```
> crc delete
> crc start
> ```
> 
> Plesse, do not run `crc cleanup` as the side-effects are that you also need to re-setup the environment.

I have done this, does not work.

**Comment by oluies** [comment]
I downgraded to 
CodeReady Containers version: 1.28.0+08de64bd
OpenShift version: 4.7.13 (embedded in executable)

It seemes to work 
I noted 
INFO Kubelet client certificate has expired, renewing it... [will take up to 8 minutes]

**Comment by praveenkumar** [comment]
> INFO Kubelet client certificate has expired, renewing it... [will take up to 8 minutes]

@oluies This is expected because bundles which are part of crc release have only 30 days valid kubelet client certs so it need to renew it. We did make some changes to handle `Failed to update pull secret on the disk: Temporary error: pull secret not updated to disk (x205)` error which should be part of next release. I am closing this issue now.

**Comment by mattpielm** [comment]
Still seeing the issue with 1.31.2 / 4.8.4 on Mac.
mpiechot@krell 1004 ~% crc version                                                                                                                                                                                             
CodeReady Containers version: 1.31.2+19538dab
OpenShift version: 4.8.4 (not embedded in executable)

Tried cleanup, delete, etc. (reset the crc config each time, even deleted ~/.crc in desperation). New pull-secret downloaded as well.

**Comment by mattpielm** [comment]
Confirmed 1.28.0 works fine, so it's a regression.
mpiechot@krell 1002 ~% crc version                                                                                                                                                                                              WARN A new version (1.31.2) has been published on https://developers.redhat.com/content-gateway/file/pub/openshift-v4/clients/crc/1.31.2/crc-macos-amd64.pkg
CodeReady Containers version: 1.28.0+08de64bd
OpenShift version: 4.7.13 (not embedded in executable)

**Comment by cfergeau** [comment]
@mattpielm since the initial issue happens on Windows, it would be very useful if you could open a new bug about this, and provide the detailed information listed in the issue template. Thanks!

**Comment by pjos** [comment]
This problem still exist in 1.32.1

```
$ crc version
CodeReady Containers version: 1.32.1+4d40571f
OpenShift version: 4.8.5 (bundle installed at C:\WS\git\bin\crc-win\crc_hyperv_4.8.5.crcbundle)
```

```
...
INFO Check DNS query from host...
INFO Adding proxy configuration to kubelet and crio service...
INFO Verifying validity of the kubelet certificates...
INFO Starting OpenShift kubelet service
INFO Waiting for kube-apiserver availability... [takes around 2min]
INFO Adding user's pull secret to the cluster...
INFO Updating SSH key to machine config resource...
INFO Waiting for user's pull secret part of instance disk...
Failed to update pull secret on the disk: Temporary error: pull secret not updated to disk (x206)
```

**Comment by cfergeau** [comment]
> This problem still exist in 1.32.1

Can you open a new issue and fill the issue template so that this an be investigated properly?

**Comment by cfergeau** [comment]
> This problem still exist in 1.32.1

If you are using a proxy setup with a CA file, see https://github.com/code-ready/crc/issues/2715#issuecomment-919123301
